changes/build-matrix.js

```diff
@@ -7,6 +7,163 @@ function run(github, context, core) {
   const allowed_build_modes = new Set(["auto", "none", "manual", "other"]);
   const other_err = 'setting as "other", which requires a fully manual scan with no automatic CodeQL analysis';
 
+  // Note: The CodeQL config filter pattern characters ?, +, [, ], ! are not supported and will be matched literal
+  // After extensive testing against JS in April 2015 - I would AVOID {} and [] globs as they are extremely limited support
+  const pathIgnoreDefaults = {
```

Comment:
Thoughts on making this list opinionated?  I like it for the OOTB config as it will be most performant. IF it ends up excluding something that is desired, a custom config file will override.

Reply:
I like having it opinionated, and as you say there is an "escape hatch" of specifying a custom CodeQL config.

It might be nice to be able to declaratively add new path excludes in the project definition, that said. Would it be much work to add this? I don't think so